src/shared/machine-image.c

```diff
@@ -223,8 +222,12 @@ static int image_make(
                 st = &stbuf;
         }
 
-        if (!path)
-                safe_getcwd(&cwd);
+        if (!path) {
+                if (dfd == AT_FDCWD)
+                        (void) safe_getcwd(&parent);
+                else
+                        (void) fd_get_path(dfd, &parent);
+        }
```

Comment:
`path` is only NULL when `dfd == AT_FDCWD`, so this path doesn't change anything as far as current callers are concerned. But it makes things safe for future uses.

Reply:
Maybe the assert should be removed?